pkg/cfn/builder/cluster.go

```diff
@@ -24,7 +25,10 @@ type ClusterResourceSet struct {
 }
 
 // NewClusterResourceSet returns a resource set for the new cluster
-func NewClusterResourceSet(provider api.ClusterProvider, spec *api.ClusterConfig, supportsManagedNodes bool) *ClusterResourceSet {
+func NewClusterResourceSet(provider api.ClusterProvider, spec *api.ClusterConfig, supportsManagedNodes bool, existingStack *gjson.Result) *ClusterResourceSet {
+	if existingStack != nil {
```

Comment:
Can this affect other commands? Like any of the `utils` commands for example? I'm worried that this mutates the ClusterConfig quite "late". Perhaps we should mutate a copy instead? Could it happen that another task that has to run later or in parallel needs the original SGs?

Reply:
>Can this affect other commands? Like any of the utils commands for example?

Only `create cluster` and `update cluster` use this type. 

> I'm worried that this mutates the ClusterConfig quite "late". Perhaps we should mutate a copy instead

This type already mutates the ClusterConfig in AddAllResources, which happens later. If we pass a copy, we'd have to use the mutated copy later in code.

> Could it happen that another task that has to run later or in parallel needs the original SGs?
The original SGs will eventually be available when the stack creation/update completes.
This does not change the behaviour of `create cluster` as the `existingStack` is nil in that case.
Update has no concurrent tasks.